### PR TITLE
Faster front end

### PR DESCRIPTION
**Pull request by eoinkelly**
This change includes a number of performance improvements. The goal is to

1. Make the pages load more quickly
1. Download the minimum amount of data to users. If users are visiting from mobile connections they pay by the MB so we want to be respectful of that.

#### JS changes

* Split JS into a bundle for the admin pages and public pages. It turns out we were loading a lot of JS for the public pages which was only required for the (logged in) admin pages. 
* Stop loading all of JQueryUI - we are now pulling in just the parts we need.

These JS changes make the JS downloaded to users much smaller - it drops from 396kB -> 96kB (compressed) which is 1.4MB -> 332kB uncompressed.

#### Font changes: 

* Allow the default fonts to display the page if _Montserrat_ font from Google fonts is loading slowly - this will cause a layout change when the real font comes in but that seems less bad than showing a white screen.

#### Video changes

* Changed `rel="auto"` to `rel="metadata"` - this stops the browser from downloading all the videos on the page in the background. A video will now be downloaded when the user clicks on it.

I have created a staging site at https://nzsl-faster-front-end-xocta7hz.herokuapp.com/ for this PR (the creds are the same as the usual staging site) so y'all can do manual testing as required.

### CODE REVIEW

**Comment by eoinkelly** [comment]
The changes to the video I made cause a slight movement of elements on the page so Percy flagged this change. I manually inspected the change and it seems fine. I'll wait for y'all to review the staging site before I merge because that's the most meaningful test.

**Comment by Br3nda** [comment]
Looks like you've also fixed the video controls regression. good work

File: app/assets/javascripts/admin_application.js
Comment (Br3nda) [inline app/assets/javascripts/admin_application.js:5]: it seems odd to me that jquery is only loaded by the admin pages...

Comment (Br3nda) [reply app/assets/javascripts/admin_application.js:5]: maybe a bunch of these can be removed, as they're already loaded in application.js

File: app/views/signs/show.html.haml
Comment (Br3nda) [inline app/views/signs/show.html.haml:52]: :+1:

File: app/assets/javascripts/admin_application.js
Comment (eoinkelly) [reply app/assets/javascripts/admin_application.js:5]: `jquery` is still loaded for the public facing pages, just not by this file.

The public facing part of the site is rendered by `app/views/layouts/application.html.haml` which loads `app/assets/javascripts/application.js` which in turn loads jquery.

The admin part of the site is rendered by `app/views/layouts/admin.html.haml` which loads `app/assets/javascripts/admin_application.js` which is this file.

**Review by Rabid-Dan — APPROVED** [review]
I dont have access to staging creds so I cant do any manual testing on the test site you've spun up but this looks like a really good set of changes to me.

File: app/assets/javascripts/admin_application.js
Comment (Br3nda) [reply app/assets/javascripts/admin_application.js:5]: I usually see this done by including application.js in admin.js.

Comment (eoinkelly) [reply app/assets/javascripts/admin_application.js:5]: In this case I think that a lot of the front-end JS isn't actually required for the admin but we are including it. I didn't bother optimizing it because basically nobody uses the admin anymore and I was running out of time.

Comment (Br3nda) [reply app/assets/javascripts/admin_application.js:5]: :elephant:

**Review by Br3nda — APPROVED** [review]

**Pull request merged**
